test/libsolidity/smtCheckerTests/special/msg_value_inheritance_1.sol

```diff
@@ -0,0 +1,17 @@
+pragma experimental SMTChecker;
+
+contract A {
+	uint public x = msg.value;
+	constructor() payable {
+		assert(x == 0); // should fail, A can be constructed with any msg.value
+	}
+}
+
+contract C is A {
+	uint public v = msg.value; // 1
+	constructor() A() {
+		assert(v == 0); // should hold
+	}
+}
+// ----
+// Warning 6328: (101-115): CHC: Assertion violation happens here.\nCounterexample:\nx = 1\n\nTransaction trace:\nA.constructor(){ value: 1 }
```

Comment:
Why is `CHC` warning in spite of the constraint being added to silence exactly this warning?

Reply:
Ah I think I understand: The warning accounts for base contract having a payable constructor.

Reply:
Yes, since the base contract has payable constructor, it can be constructed using any `msg.value`, so the assertion is violated when the contract is constructed with value greater than 0.